conan/tools/cmake/toolchain/blocks.py

```diff
@@ -328,6 +328,42 @@ class AppleSystemBlock(Block):
         # Setting CMAKE_OSX_DEPLOYMENT_TARGET if "os.version" is defined by the used conan profile
         set(CMAKE_OSX_DEPLOYMENT_TARGET "{{ CMAKE_OSX_DEPLOYMENT_TARGET }}" CACHE STRING "")
         {% endif %}
+        set(BITCODE "")
+        set(FOBJC_ARC "")
+        set(VISIBILITY "")
+        {% if enable_bitcode %}
+        # Bitcode ON
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE "YES")
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE "bitcode")
+        {% if enable_bitcode_marker %}
+        set(BITCODE "-fembed-bitcode-marker")
+        {% else %}
+        set(BITCODE "-fembed-bitcode")
+        {% endif %}
+        {% elif enable_bitcode is not none %}
+        # Bitcode OFF
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE "NO")
+        {% endif %}
+        {% if enable_arc %}
+        # ARC ON
+        set(FOBJC_ARC "-fobjc-arc")
+        set(CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC "YES")
+        {% elif enable_arc is not none %}
+        # ARC OFF
+        set(FOBJC_ARC "-fno-objc-arc")
```

Comment:
I am still a bit confused about the duplication, of setting the raw flags ``"-fno-objc-arc"``, and also the CMake ``CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC`` variable. Isn't that duplicating the flags?

Reply:
Ok, I see it is because one thing is useful only for the Xcode generator.
The original iostoolchain did that, so better use the Cmake generator to only define the flags when needed (not Xcode), and only define the CMAKE_XCODE_ATTR... when needed.

Reply:
Check this condition: https://github.com/leetal/ios-cmake/blob/14092d317b84a3037e71b870d2117dda126811a3/ios.toolchain.cmake#L779

Reply:
these are compiler flags, they aren't specific to Xcode somehow. could be used in autotools, CMake, Meson, etc.

Reply:
@SSE4 yes, that's the point, but the original toolchain sets the `CMAKE_XCODE_ATTR...` when the generator is xcode OTHERWISE it sets the compiler flags. Here in this PR everything is set, so it will be duplicated somehow.